backend/models/models.go

```diff
@@ -14,12 +14,42 @@
 
 package models
 
+type GeneralPermissions struct {
```

Comment:
Can you add some inline comments to help understand what the struct fields are used for? For example, it's not very clear to me what's the purpose of AddTeamPermissions in TeamList.

Reply:
Let me know if the comments make it a bit clearer now.

Reply:
The comments definitely help, though I think `AddTeamPermissions` would be better named something like `HasAddTeamPermissions` or `CanAddTeams`.

Normally boolean-typed things are best given names starting with a predicate verb like "is", "can", or "has".

Reply:
👍🏻